### PR TITLE
Light support cull mask

### PR DESCRIPTION
**Pull request by GuoLei1990**
### Please check if the PR fulfills these requirements

- [ ] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### Other information:

### CODE REVIEW

File: packages/core/src/Renderer.ts [hunk 1]
```diff
@@ -71,6 +72,9 @@ export class Renderer extends Component {
   @assignmentClone
   private _receiveShadows: boolean = true;
 
+  @assignmentClone
```
Comment (zhuxudong) [inline packages/core/src/Renderer.ts:75]: `deepClone`

File: packages/core/src/lighting/DirectLight.ts [hunk 2]
```diff
@@ -21,6 +23,7 @@ export class DirectLight extends Light {
   static _updateShaderData(shaderData: ShaderData): void {
     const data = DirectLight._combinedData;
 
+    shaderData.setIntArray(DirectLight._cullingMaskProperty, data.cullingMask);
```
Comment (zhuxudong) [inline packages/core/src/lighting/DirectLight.ts:26]: why not use `setFloat` instead using 2 mask in IntArray?

File: packages/core/src/Renderer.ts (hunk 1 above)
Comment (GuoLei1990) [reply packages/core/src/Renderer.ts:75]: ignoreClone is right

File: packages/core/src/lighting/DirectLight.ts (hunk 2 above)
Comment (GuoLei1990) [reply packages/core/src/lighting/DirectLight.ts:26]: https://www.khronos.org/registry/OpenGL/specs/es/2.0/GLSL_ES_Specification_1.00.pdf

Integer precision qualifiers specify the desired precision of integer data types used in the declaration of variables. They specify a minimum range and precision supported by the implementation for each precision qualifier. The available integer precision qualifiers are:

highp: represents high precision, supports at least 16 bits of integer precision.
mediump: represents medium precision, supports at least 10 bits of integer precision.
lowp: represents low precision, supports at least 8 bits of integer precision.

File: packages/core/src/Renderer.ts [hunk 3]
```diff
@@ -26,6 +26,7 @@ export class Renderer extends Component {
   private static _mvpMatrixProperty = Shader.getPropertyByName("u_MVPMat");
   private static _mvInvMatrixProperty = Shader.getPropertyByName("u_MVInvMat");
   private static _normalMatrixProperty = Shader.getPropertyByName("u_normalMat");
+  private static _rendererLayerProperty = Shader.getPropertyByName("oasis_RendererLayer");
```
[the remark this answers is not in the document]
Comment (GuoLei1990) [reply packages/core/src/Renderer.ts:29]: No! Oasis internal named rule is refer to `gl_FragColor`

File: packages/core/src/lighting/DirectLight.ts (hunk 2 above)
Comment (GuoLei1990) [reply packages/core/src/lighting/DirectLight.ts:26]: To be honest, I don't want to be compatible with webgl1.0!

File: packages/core/src/Renderer.ts (hunk 1 above)
Comment (GuoLei1990) [reply packages/core/src/Renderer.ts]: done

**Review by zhuxudong — APPROVED** [review]

**Pull request merged**